FAQ.md

```diff
@@ -0,0 +1,104 @@
+
+If the machine’s status does not have any instance ID or networking information associated with it, most likely the instance has failed to be created.  This can be due to misconfiguration, inadequate or invalid cloud credentials, cloud API quota exhaustion, or other cloud API problems such as an outage or temporary network condition.  You will need to inspect the machine-controllers logs for more definitive information.
+
+## What happens when I delete a Machine?
+This section assumes you’ve deleted a machine from the API, rather than deleting a VM or Instance in the cloud provider.
+
+When a machine is marked deleted, the associated node is immediately cordoned and drained.  Draining utilizes the eviction API, so PodDisruptionBudgets will be respected.  If a node cannot be successfully drained due to a PDB, the operation will retry indefinitely.
+
+After an associated node is drained, the associated instance or VM is deleted from the cloud provider.  After the instance is deleted in the cloud provider, the associated node object is deleted from the API.
+
+Finally, the finalizer is removed from the machine object and the machine object is removed from the API.
+
+## I want to skip draining when I delete a Machine
+This is not recommended for most cases, especially Master Machines.  Properly draining machines will respect PodDisruptionBudgets and prevent the cluster and workloads from going into an unhealthy state.
+
+You can optionally set an **annotation** **"machine.openshift.io/exclude-node-draining"** on each machine object you wish for draining to be skipped.  This can be applied or removed at any time.
+
+## What happens if I delete an Instance or VM outside of the Machine API, such as in the AWS web console?
+This is not recommended.  By default, the machine-api will not take any corrective action.  If you are utilizing MachineHealthChecking, the machine may get deleted depending on the configuration of the MHC.
+
+## Can I remove the finalizer for a machine that is stuck in deleting?
+This is not recommended.  This may result in orphaned node objects and orphaned compute resources.
+
+## How do I set a Node’s Role (eg, Worker)
+This is set via the MCO’s config, not the machine-api. The config a machine receives upon boot is decided by the machine's user-data.
+
+## How does the user-data get created for the VMs?
+The user-data is generated by the installer.  The machine-api is just a consumer of the user-data and simply passes the data to the cloud.  Refer to documentation for the machine-config-operator (not part of the machine-api) for creating new machine pools and user-data.
+
+## Adding Labels to Nodes via Machines
+Labels are only added to nodes via machines at creation time.  You can add arbitrary labels which are applied to nodes immediately after they join the cluster.  Adding subsequent labels on the machine object (or in a machineset’s template) will have no effect.
```

Comment:
may be add the api path to the label here, that's usually confusing as there're multiple labels